autogluon/utils/tabular/ml/models/fastainn/tabular_nn_fastai.py

```diff
@@ -0,0 +1,334 @@
+
+        self.model = tabular_learner(
+            data, layers=layers, ps=ps, emb_drop=self.params['emb_drop'], metrics=nn_metric,
+            loss_func=loss_func, callback_fns=[early_stopping_fn]
+        )
+        logger.log(0, self.model.model)
+
+        with make_temp_directory() as temp_dir:
+            save_callback = SaveModelCallback(self.model, monitor=objective_func_name_to_monitor, mode=objective_optim_mode, name=self.name)
+            with progress_disabled_ctx(self.model) as model:
+                original_path = model.path
+                model.path = Path(temp_dir)
+                model.fit_one_cycle(self.params['epochs'], self.params['lr'], callbacks=save_callback)
+
+                # Load the best one and export it
+                model.load(self.name)
+
+                if objective_func_name == 'log_loss':
+                    self.eval_result = model.validate()[0]
+                else:
+                    self.eval_result = model.validate()[1].numpy().reshape(-1)[0]
+
+                logger.log(0, f'Model validation metrics: {self.eval_result}')
+                model.path = original_path
+
+    def _generate_datasets(self, X_train, Y_train, X_val, Y_val):
+        df_train = pd.concat([X_train, X_val], ignore_index=True)
+        df_train[LABEL] = pd.concat([Y_train, Y_val], ignore_index=True)
+        train_idx = np.arange(len(X_train))
+        val_idx = np.arange(len(X_val)) + len(X_train)
+        return df_train, train_idx, val_idx
+
+    def __get_objective_func_name(self):
+        from fastai.metrics import root_mean_squared_error, mean_squared_error, mean_absolute_error, accuracy, FBeta, AUROC, Precision, Recall, r2_score
+
+        metrics_map = {
+            # Regression
+            'root_mean_squared_error': root_mean_squared_error,
+            'mean_squared_error': mean_squared_error,
+            'mean_absolute_error': mean_absolute_error,
+            'r2': r2_score,
+            # Not supported: median_absolute_error
+
+            # Classification
+            'accuracy': accuracy,
+
+            'f1': FBeta(beta=1),
+            'f1_macro': FBeta(beta=1, average='macro'),
+            'f1_micro': FBeta(beta=1, average='micro'),
+            'f1_weighted': FBeta(beta=1, average='weigthed'),  # this one has some issues
+
+            'roc_auc': AUROC(),
+
+            'precision': Precision(),
+            'precision_macro': Precision(average='macro'),
+            'precision_micro': Precision(average='micro'),
+            'precision_weigthed': Precision(average='weigthed'),
+
+            'recall': Recall(),
+            'recall_macro': Recall(average='macro'),
+            'recall_micro': Recall(average='micro'),
+            'recall_weigthed': Recall(average='weigthed'),
+            # Not supported: pac_score
+        }
+
+        objective_func_name = self.eval_metric.name
```

Comment:
self.stopping_metric.name , eval_metric is only ever used for after-training scoring.

Reply:
> One concern before we add it into our default-set of models would be ensuring the `verbosity` argument of `task.fit()` can control the amount of printed output from this model (right now it's printing out too much output at the default verbosity-level)...

I updated the logging levels in our code, but few things are logged from fast.ai code and changing that would require patching.